core/controlplane/config/templates/cloud-config-controller

```diff
@@ -1362,7 +1362,7 @@ write_files:
                   - /bin/sh
                   - -xec
                   - |
-                    metadata() { wget -O - -q http://169.254.169.254/2016-09-02/"$1"; }
+                    metadata() { curl -s -f http://169.254.169.254/2016-09-02/"$1"; }
```

Comment:
`-s` also suppresses any error message, which seems undesirable.
How about `--silent --show-error` instead so that it still prints error messages?

```
$ curl --fail --silent http://httpstat.us/502
$ echo $?
22
```

```
$ curl --fail --silent --show-error http://httpstat.us/502
curl: (22) The requested URL returned error: 502 Bad Gateway
$ echo $?
22
```

Reply:
@mumoshu sure I've done that. I've kept the short flags to be consistent with the rest of the file.

There are many other curls in this file that silence but do not have the show error flag, probably we should change them all. There's many without the specific fail flag too which is concerning (some valid when we want to check HTTP response code).

Reply:
Thanks! Yeah, short flags would be better for consistency.

> There are many other curls in this file that silence but do not have the show error flag, probably we should change them all.

Can't agree more.
Would you mind fixing these issues too, in an another PR? 🙇